src/components/ui/dashboard/coaching-session-dialog.tsx

```diff
@@ -0,0 +1,150 @@
+import { Label } from "@/components/ui/label";
+import { Calendar } from "@/components/ui/calendar";
+import { useCoachingRelationshipStateStore } from "@/lib/providers/coaching-relationship-state-store-provider";
+import { getDateTimeFromString } from "@/types/general";
+import {
+  CoachingSession,
+  defaultCoachingSession,
+} from "@/types/coaching-session";
+import {
+  useCoachingSessionList,
+  useCoachingSessionMutation,
+} from "@/lib/api/coaching-sessions";
+import { DateTime } from "ts-luxon";
+import { useAuthStore } from "@/lib/providers/auth-store-provider";
+import { cn } from "@/components/lib/utils";
+import { format } from "date-fns";
+
+interface CoachingSessionDialogProps {
+  open: boolean;
+  onOpenChange: (open: boolean) => void;
+  onCoachingSessionUpdated: () => void;
+  dialogTrigger?: React.ReactElement<React.HTMLAttributes<HTMLButtonElement>>;
+  existingSession?: CoachingSession;
+  mode: "create" | "update";
+}
+
+export function CoachingSessionDialog({
+  open,
+  onOpenChange,
+  onCoachingSessionUpdated,
+  dialogTrigger,
+  existingSession,
+  mode,
+}: CoachingSessionDialogProps) {
+  const { currentCoachingRelationshipId } = useCoachingRelationshipStateStore(
+    (state) => state
+  );
+  const fromDate = DateTime.now().minus({ month: 1 });
+  const toDate = DateTime.now().plus({ month: 1 });
+  const { refresh } = useCoachingSessionList(
+    currentCoachingRelationshipId,
+    fromDate,
+    toDate
+  );
+  const { create: createCoachingSession, update } =
+    useCoachingSessionMutation();
+  const [sessionDate, setSessionDate] = useState<Date | undefined>(
+    existingSession ? new Date(existingSession.date) : undefined
+  );
+  const [sessionTime, setSessionTime] = useState<string>(
+    existingSession ? format(new Date(existingSession.date), "HH:mm") : ""
+  );
+  const { isCoach } = useAuthStore((state) => state);
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+    if (!sessionDate || !sessionTime) return;
+
+    const [hours, minutes] = sessionTime.split(":").map(Number);
+    const dateTime = getDateTimeFromString(sessionDate.toISOString())
+      .set({ hour: hours, minute: minutes })
+      .toFormat("yyyy-MM-dd'T'HH:mm:ss");
+
+    try {
+      if (mode === "create") {
+        const newCoachingSession: CoachingSession = {
+          ...defaultCoachingSession(),
+          coaching_relationship_id: currentCoachingRelationshipId,
+          date: dateTime,
+        };
+        await createCoachingSession(newCoachingSession);
+      } else if (existingSession) {
+        await update(existingSession.id, {
+          ...existingSession,
+          date: dateTime,
+          updated_at: DateTime.now(),
+        });
+      }
+      refresh();
+      onCoachingSessionUpdated();
+      setSessionDate(undefined);
+      setSessionTime("");
+      onOpenChange(false);
+    } catch (error) {
+      console.error(`Failed to ${mode} coaching session:`, error);
```

Comment:
Would you mind adding a TODO comment here saying to also display an error as a toast element to the user once we get toast infrastructure in place?

Reply:
added! e0074c6281903cfd0496013ff251999b56c24be7